misaligned.cpp

```diff
@@ -1,21 +1,47 @@
+#include "./misaligned.h"
 #include <assert.h>
 #include <iostream>
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
-    for (i = 0; i < 5; i++) {
-        for (j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
-        }
+
+void testColorMap() {
+    const char* majorColor[] = {"White", "Red", "Black"};
+    const char* minorColor[] = {"Blue", "Orange"};
+    auto colorMap = createColorMap(majorColor, 3, minorColor, 2);
+    assert(colorMap.size() == 6);
+    for (auto pair : colorMap) {
+        assert(pair.first == "White" || pair.first == "Red" || pair.first == "Black");
+        assert(pair.second == "Blue" || pair.second == "Orange");
     }
-    return i * j;
 }
 
-int main() {
-    int result = printColorMap();
+void testPrintContent() {
+    std::vector<std::pair<std::string, std::string>> colorMap{
+        std::make_pair("White", "Blue"),
+        std::make_pair("White", "Orange"),
+        std::make_pair("Red", "Blue"),
+        std::make_pair("Red", "Orange")};
+    auto generatedContent = generatePrintContent(colorMap);
+    assert(generatedContent.size() == 4);
+    for (auto content : generatedContent) {
+        std::cout << content << std::endl;
+    }
+}
+
+void testPrintColorMap() {
+    auto result = printColorMap();
     assert(result == 25);
+}
+
+void testPrintColorMapVisually(bool visuallyOk) {
+    assert(visuallyOk);
+}
+
+int main() {
+    testColorMap();
+    testPrintContent();
+    testPrintColorMap();
+    // pass true when visually inspecting the output and its good
+    testPrintColorMapVisually(false);

```

Comment:
This leaves a bit of subjectivity in the test... a human viewer will need to decide what is visually ok.
That requires experience and knowledge, which could be used in more productive work.
Think of what aspects make it 'visually appealing'? Is it vertical alignment of the column separators? Presence of a header? etc. Then try to assert them.

Reply:
Addressed the comment. There is one more tests needed to check index number alignment and header which is todo for now.